src/libraries/System.Net.Sockets/tests/FunctionalTests/TcpListenerTest.cs

```diff
@@ -44,6 +44,30 @@ public void Active_TrueWhileRunning(int ctor)
             Assert.False(listener.Active);
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows))]
+        public void AllowNatTraversal_NotStarted_SetSuccessfully()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+            listener.AllowNatTraversal(true);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows))]
+        public void AllowNatTraversal_Started_ThrowsException()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+            listener.Start();
+            Assert.Throws<InvalidOperationException>(() => listener.AllowNatTraversal(true));
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows))]
```

Comment:
```suggestion
        [Fact]
        [PlatformSpecific(TestPlatforms.Windows)]
```

Reply:
I added that condition because [tests failed on Unix, Linux and MacOS](https://dev.azure.com/dnceng/public/_build/results?buildId=529779&view=results) with ```System.PlatformNotSupportedException : IP protection level cannot be controlled on this platform.```

Reply:
I know.  That's why I added the PlatformSpecific attribute instead.  The Conditional attribute is for more complex constraints that can't be expressed as just the Windows/Linux/macOS target.

Reply:
Ah, I see. Sorry, I misread the diff.